discovery-infra/start_discovery.py

```diff
@@ -291,6 +303,84 @@ def validate_dns(client, cluster_id):
         raise e
 
 
+def configure_terraform_workers_nodes(tf_folder, num_worker_nodes):
+    tfvars = utils.get_tfvars(tf_folder)
+    num_workers = tfvars['worker_count']
+    num_workers = num_workers + num_worker_nodes
+    tfvars['worker_count'] = num_workers
+    master_end_ip = tfvars['libvirt_master_ips'][-1]
+    workers_ip_list = tfvars['libvirt_worker_ips']
+    if not workers_ip_list:
+        worker_starting_ip = ipaddress.ip_address(master_end_ip)
+    else:
+        worker_starting_ip = ipaddress.ip_address(tfvars['libvirt_worker_ips'][0])
+    worker_ips_list = workers_ip_list + _create_ip_address_list(num_worker_nodes, worker_starting_ip + 1)
+    tfvars['libvirt_worker_ips'] = worker_ips_list
+    utils.set_tfvars(tf_folder, tfvars)
+
+
+def configure_terraform_api_dns(tf_folder, api_vip_ip, api_vip_dnsname):
+    main_str = utils.get_tf_main(tf_folder)
+    start_index = main_str.find("dns {\n    ") + len("dns {\n    ")
+    dns_config = "hosts  {\n      ip = \"%s\"\n      hostname = \"%s\"\n    }\n    " % (api_vip_ip, api_vip_dnsname)
+    main_str = main_str[:start_index] + dns_config + main_str[start_index:]
+    utils.set_tf_main(tf_folder, main_str)
+
+def get_network_nodes_from_terraform(tf_folder):
+    tfvars = utils.get_tfvars(tf_folder)
+    return tfvars['libvirt_network_name'], tfvars['master_count'] + tfvars['worker_count']
+
+
+def apply_day2_tf_configuration(cluster_name, num_worker_nodes, api_vip_ip, api_vip_dnsname):
+    tf_folder = utils.get_tf_folder(cluster_name, args.namespace)
+    configure_terraform_workers_nodes(tf_folder, num_worker_nodes)
+    configure_terraform_api_dns(tf_folder, api_vip_ip, api_vip_dnsname)
+    return get_network_nodes_from_terraform(tf_folder)
+
+
+def day2_nodes_flow(client, cluster_name, cluster, image_path, num_worker_nodes, api_vip_ip, api_vip_dnsname):
+    tf_network_name, total_num_nodes = apply_day2_tf_configuration(cluster_name, num_worker_nodes, api_vip_ip, api_vip_dnsname)
+    utils.run_command(f'make _apply_terraform CLUSTER_NAME={cluster_name}')
```

Comment:
lets have utils.run_command(f'make _apply_terraform CLUSTER_NAME={cluster_name}') in utils and use it from 2 places.
i think u must have filelock on it @RazRegev ?

Reply:
@RazRegev why do we need the filelock? each terraform is in a different directory defined by cluster and namespace